internal/cluster/connection_manager.go

```diff
@@ -463,11 +462,13 @@ func (m *ConnectionManager) connectCluster(ctx context.Context, cluster *Candida
 	return initialAddr, nil
 }
 
-func (m *ConnectionManager) ensureConnection(ctx context.Context, addr pubcluster.Address) (*Connection, error) {
+func (m *ConnectionManager) ensureConnection(ctx context.Context, addr pubcluster.Address, networkCfg *pubcluster.NetworkConfig) (*Connection, error) {
 	conn := m.createDefaultConnection()
-	if err := conn.start(m.clusterConfig, addr); err != nil {
+	err := conn.start(networkCfg, addr)
+	if err != nil {
```

Comment:
Can we merge this into one line with the statement above?

Reply:
Updated at: https://github.com/hazelcast/hazelcast-go-client/pull/797/commits/f1f54ae444f68d631c89d6634699f10ff0898a1f